packages/core/src/useEventEmitter.tsx

```diff
@@ -24,6 +24,7 @@ export default function useEventEmitter<T extends Record<string, any>>(
   const listeners = React.useRef<Record<string, Record<string, Listeners>>>({});
 
   const create = React.useCallback((target: string) => {
+    let removed = false;
```

Comment:
This should be stored inside the closure of the unsubscribe function so that it only applies to individual `addListener` calls. so like:

```js
return () => {
  let removed = false;
  if (!removed) {
    removed = true;
    removeListener(type, callback);
  }
}
```

Reply:
The `removed` variable had to be in the body of `addListener` in order to fix the issue. Fixed in [dce1b69](https://github.com/react-navigation/react-navigation/pull/10667/commits/dce1b69df56a06b85f45dcc4ed48865120dcc26b)